miio/vacuum_tui.py

```diff
@@ -0,0 +1,95 @@
+try:
+    import curses
+except ImportError:
+    pass
```

Comment:
When can this happen?

Reply:
What we import here as `curses` is actually an extension module that must be backed by one of few C implementations of a `curses` library. `ImportError` occurs when none of the implementations are available on the system.

I do not think there is much we can do about it except for mentioning in `docs/vacuum.rst`, that `curses` lib is a requirement for TUI.

Reply:
Ah, I see. I thought it is part of the standard library and always available.

How about letting it bubble up and doing the import inside the `tui()` command? It could be caught there to display a user-friendly error message.

Reply:
Personally, I prefer all the `import`s to be grouped in one place - at the top of a module. They are easier to manage that way.

If all we want is a prettier error message, I would rather delay the exception until `VacuumTUI` is instantiated:

```python
try:
    import curses
except ImportError:
    curses = None

...

class VacuumTUI:
    def __new__(cls, *args, **kwargs):
        if curses is None:
            raise ImportError("curses library is not available")
        return super().__new__(cls)
```

What do you think?

Reply:
So what I want is it not to simply crash silently in any case, so your suggested solution works fine, too! Maybe check for the Noneness inside the `__init__` though?

Reply:
Moved the check to `__init__` in a new commit.